addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml

```diff
@@ -70,6 +70,10 @@
                     hasMarkAsReadButton="preview.hasMarkAsReadButton"
                     isLast="preview_last and failures.length === 0"
                     onClick="(isMarkAsRead) => this.onClickPreview(isMarkAsRead, preview)"
+                    onSwipeRight="() => this.markAsRead(preview)"
```

Comment:
for needaction it should mark it as read, for preview it should set last seen message

Reply:
That's what it does: either `set_last_seen_message` or `mark_all_as_read`.

Reply:
method name is very confusing, but ok for this PR